### DISCUSSION TITLE
Environment management across Unix users

### DISCUSSION THREAD

**Discussion by Lookszz** [python-poetry/poetry#6303] [General] [answered]
- [ x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

So I'm having a problem for quite some time now using Poetry package managing. I have a Django project where I use Poetry for package managing. All of a sudden adding packages didn't work anymore. I would get error messages saying that the packages didn't exist. I found the problem, namely that poetry was linked to another virtualenv than my project:

project: /home/vfs/.cache/pypoetry/virtualenvs/vfs-gOLk6wo--py3.7
poetry: /root/.cache/pypoetry/virtualenvs/vfs-gOLk6wo--py3.7

I tried changing the poetry config values to the project values. But when doing this and running 'poetry install', it says the packages are installing and I get no errors. But the packages won't get into my virtualenv. And when running 'poetry install' again it will do the same installs. What should I do here?

**Comment by Secrus** [comment]
What Poetry version are you using and how was it installed?

**Comment by Lookszz** [comment]
> What Poetry version are you using and how was it installed?

So I took over this project and wasn't involved in the setup. But the current version of Poetry is:
Poetry version 1.0.4

In my uwsgi.ini the path is: /home/vfs/.cache/pypoetry/virtualenvs/vfs-gOLk6wo--py3.7
So I just need to change the poetry path, which is not working for some reason

**Comment by Lookszz** [comment]
@Secrus I upgraded to the newest version (1.1.15) and it is still not working.

**Comment by neersighted** [comment]
You are running Poetry as root and the webserver as the `vfs` user or something similar. Poetry isn't doing anything wrong here -- environments are explicitly scoped to the current user.

The best work-around here is to `poetry config --local virtualenvs.in-project true` and then Poetry will install to `<path to my project>/.venv` -- point your UWSGI server at that path and you'll be golden.

**Comment by Lookszz** [comment]
> You are running Poetry as root and the webserver as the `vfs` user or something similar. Poetry isn't doing anything wrong here -- environments are explicitly scoped to the current user.
> 
> The best work-around here is to `poetry config --local virtualenvs.in-project true` and then Poetry will install to `<path to my project>/.venv` -- point your UWSGI server at that path and you'll be golden.

Still doesn't work.... It states that it installs/updates all packages. But when I run poetry install again it does it again. Current set up (uwsgi):
home                = /home/vfs/.cache/pypoetry/virtualenvs/vfs-gOLk6wo--py3.7

poetry:
cache-dir = '/home/vfs/.cache/pypoetry'
experimental.new-installer = true
installer.parallel = true
virtualenvs.create = true
virtualenvs.in-project = true
virtualenvs.path = '/home/vfs/.cache/pypoetry/virtualenvs'

Maybe I can try to remove the virtualenv? And do a fresh poetry install, or is this a bad idea?

**Comment by neersighted** [comment]
Sorry, I thought 'start over clean' was implicit; try starting clean with an in-project venv (or a manually created venv that you activate -- Poetry will respect it) and you should be able to get things working.

To reiterate again -- Poetry cannot discover the venv you want it to use as it is being run as two different users, so you need to do something about it (run as the same user, use an in-project venv, externally manage the venv), etc.

**Reply by Lookszz** [reply]
Okay so because I run as 'root@vfs' poetry thinks the virtualenv is in the /root/.cache/virtualenvs/ directory? Even when my settings are pointing to '/home/vfs/.cache'? Because when I go into that virtualenv (/root/...) and look at the packages everything looks good. Sorry I have little experience server wise. Can you explain the best solution in beginner language?

**Reply by neersighted** [reply]
Essentially yes. While technically it should be possible to do what you're trying to do, it's way to hard to debug over the internet.

Instead, I would suggest either creating an environment with `python -m venv /path/to/my/venv` and activating it so that Poetry will use it, or setting `poetry config virtualenvs.in-project true` so that Poetry will create the env at `<project root>/.venv`.

**Reply by Lookszz** [reply]
If I do: 
- source /home/vfs/.cache/pypoetry/virtualenvs/vfs-gOLk6wo--py3.7/bin/activate 
- go into directory with .lock files etc (cd /home/vfs/public_html/vfs)
- poetry install

After this it still doesn't work.

**Comment by Lookszz** [comment, marked as answer]
Okay I found a solution: removing the virtualenvs and installing them again 'poetry install' fixed it for me for some reason.
